### PR TITLE
PYTHON-3131 Test Failure - test_mypy on macos + auth

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: test/test_mypy.py [hunk 1]
```diff
@@ -58,8 +61,9 @@ class TestPymongo(unittest.TestCase):
     coll: Collection
 
     @classmethod
+    @client_context.require_connection
     def setUpClass(cls) -> None:
-        cls.client = MongoClient(serverSelectionTimeoutMS=250, directConnection=False)
+        cls.client = rs_or_single_client(serverSelectionTimeoutMS=250, directConnection=False)
```
Comment (ShaneHarvey) [inline test/test_mypy.py:66]: Let's make this class inherit from IntegrationTest, then we get a `self.client` and `client_context.require_connection` for free.

File: test/test_mypy.py [hunk 2]
```diff
@@ -70,6 +74,7 @@ def setUpClass(cls) -> None:
     def tearDownClass(cls) -> None:
         cls.client.close()
 
+    @client_context.require_no_auth
```
Comment (ShaneHarvey) [inline test/test_mypy.py:77]: is `client_context.require_no_auth` needed in this tests?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: test/test_mypy.py (hunk 2 above)
Comment (blink1073) [reply test/test_mypy.py:77]: Yes, it was failing without it

File: test/test_mypy.py (hunk 1 above)
Comment (blink1073) [reply test/test_mypy.py]: Done

File: test/test_mypy.py (hunk 2 above)
Comment (blink1073) [reply test/test_mypy.py]: Trying without it on this run now that we're using `IntegrationTest`

Comment (blink1073) [reply test/test_mypy.py]: Cool, not needed anymore

**Comment by ShaneHarvey** [comment]
Do the tests now pass on macos in EVG?

**Comment by blink1073** [comment]
Yes, the previously failing test is now [passing](https://spruce.mongodb.com/task/mongo_python_driver_test_macos__platform~macos_1014_auth~noauth_ssl~nossl_test_3.6_replica_set_patch_09f8aa9928e9763e790a7fd6e30bccb701efea9d_620d6da530661530d8ad91a7_22_02_16_21_33_26/tests?execution=0&sortBy=STATUS&sortDir=ASC).

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**